lib/spack/spack/cmd/install.py

```diff
@@ -31,10 +32,64 @@
 level = "short"
 
 
+def parse_use_buildcache(args):
+    bc_parsed = {}
+    bc_keys = ["package", "dependencies", ""]
+    bc_values = ["only", "never", "auto"]
+    kv_list = re.findall("([a-z]+)?:?([a-z]+)", args.use_buildcache)
```

Comment:
I'm not sure this regex is doing what we want.  If I say:

```
spack install --no-check-signature --use-buildcache auto readline
```

then spack errors with:

```
==> Error: Unrecognized arguments passed to use-buildcache: [('aut', 'o')]
==> Error: Expected: --use-buildcache :{[auto|only|never],[[package:[auto|only|never]],[dependencies:[auto|only|never]]]
```

Reply:
I love regex! That is supposed to evaluate to `("", "auto")`, which it does for `never` and `only`...

Reply:
`([a-z]+:)?([a-z]+)` works, but requires a more complicated filtering